### ISSUE TITLE
Incorrect bottom padding when removing labels

### ISSUE THREAD

**Issue by fredpyo** [masayuki0812/c3#1093]
Hello,

When hiding labels I noticed that Pie charts have a weird bottom padding that was preventing them from taking up the pixel perfect space I reserved for them.

After looking into the code I discovered this line in `core.js`:
```javascript
c3_chart_internal_fn.updateSizes = function () {
  ...
  $$.arcHeight = $$.height - ($$.isLegendRight ? 0 : 10);
  ...
}
```

When hiding labels this line still produces a 10px vertical padding... So I hacked a quick and dirty:

```javascript
$$.arcHeight = $$.height - ($$.config.legend_hide == true || $$.config.legend_show == false || $$.isLegendRight ? 0 : 10);
```

And it worked wonders. It's not a fancy fix by any means. Is there some setting I'm forgetting, or should we look for a proper fix for this issue?

Thanks!

**Comment by shrewdlogarithm** [comment]
I ran into this tonight too - you're stuck with 10px margin for a legend (width or height depending on position) even if you disable legends!

Pie charts still don't QUITE use all the available space even with that 10px removed - there's a mandatory 4px top margin for some reason and even with expand and labels disabled there's space around the chart I've not found yet???

**Comment by fredpyo** [comment]
Yes, I've noticed that extra spacing too. I imagined it was used to make space for the pie segments that expand when hovered on. A scaling factor is used to shrink the graph a little bit for these cases. Unfortunately, it seems that the scaling factor is always applied even when the hover behavior is disabled. This would also be a quick fix too though.
